geom_mgr/CubedSphere/CubedSphereGeomSpec_smod.F90

```diff
@@ -0,0 +1,290 @@
+      consistent = (ifound .eq. 3) .or. (ifound .eq. 0)
+      _ASSERT(consistent, "specfied partial stretch parameters")
+      if (.not. is_stretched) then
+         schmidt_parameters%stretch_factor = undef_schmidt 
+         schmidt_parameters%target_lon= undef_schmidt 
+         schmidt_parameters%target_lat= undef_schmidt 
+      end if
+      _RETURN(_SUCCESS)
+
+   end function make_SchmidtParameters_from_metadata
+
+   function return_r8(file_metadata, attr_name, rc) result(param)
+      real(kind=ESMF_KIND_R8) :: param
+      type(FileMetadata), intent(in) :: file_metadata
+      character(len=*), intent(in) :: attr_name
+      integer, optional, intent(out) :: rc
+ 
+      integer :: status
+      class(*), pointer :: attr_val(:)
+      type(Attribute), pointer :: attr
+
+      attr => file_metadata%get_attribute(attr_name)
+      attr_val => attr%get_values()
+      select type(q=>attr_val)
+      type is (real(kind=REAL32))
+         param = q(1)
+      type is (real(kind=REAL64))
+         param = q(1)
+      class default
+         _FAIL('unsupported subclass for stretch parameters')
+      end select
+      _RETURN(_SUCCESS)
+   end function return_r8
+      
+
+   ! Accessors
+   pure module function get_decomposition(spec) result(decomposition)
+      type(CubedSphereDecomposition) :: decomposition
+      class(CubedSphereGeomSpec), intent(in) :: spec
+
+      decomposition = spec%decomposition
+   end function get_decomposition
+
+   pure module function get_im_world(spec) result(im_world)
+      integer :: im_world
+      class(CubedSphereGeomSpec), intent(in) :: spec
+
+      im_world = spec%im_world
+   end function get_im_world
+
+   pure module function get_schmidt_parameters(spec) result(schmidt_parameters)
+      type(ESMF_CubedSphereTransform_Args) :: schmidt_parameters
+      class(CubedSphereGeomSpec), intent(in) :: spec
+
+      schmidt_parameters = spec%schmidt_parameters
+   end function get_schmidt_parameters
+
+   logical module function supports_hconfig_(this, hconfig, rc) result(supports)
+      class(CubedSphereGeomSpec), intent(in) :: this
+      type(ESMF_HConfig), intent(in) :: hconfig
+      integer, optional, intent(out) :: rc
+
+      integer :: status
+      character(:), allocatable :: geom_class
+
+      ! Mandatory entry: "class: CubedSphere"
+      supports = ESMF_HConfigIsDefined(hconfig, keystring='class', _RC)
+      _RETURN_UNLESS(supports)
+
+      geom_class = ESMF_HConfigAsString(hconfig, keyString='class', _RC)
+      supports = (geom_class == 'CubedSphere')
+      _RETURN_UNLESS(supports)
+      
+      _RETURN(_SUCCESS)
+   end function supports_hconfig_
+
+   logical module function supports_metadata_(this, file_metadata, rc) result(supports)
+      class(CubedSphereGeomSpec), intent(in) :: this
+      type(FileMetadata), intent(in) :: file_metadata
+      integer, optional, intent(out) :: rc
+
+      integer :: status
+
+      supports = file_metadata%has_dimension("Xdim", _RC)
```

Comment:
I'm slightly concerned that this is a bit too lax, at least in theory.  But I'm sure it is fine until we hit the exception.

Reply:
yes, it should have been at least nf. I'll clean this up later